### PR TITLE
Feat/auth

### PR DESCRIPTION
**Pull request by ShivCodeP**
Basic Authenication using express-session

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,17 +4,24 @@
   "main": "index.js",
   "license": "MIT",
   "dependencies": {
+    "bcrypt": "^5.0.1",
+    "body-parser": "^1.20.0",
+    "dotenv": "^16.0.1",
     "express": "^4.18.1",
+    "express-session": "^1.17.2",
+    "jsonwebtoken": "^8.5.1",
```
Comment (Sparkenstein) [inline package.json]: remove jwt

File: package.json [hunk 2]
```diff
@@ -4,17 +4,24 @@
...
     "mongoose": "^6.3.3",
     "ts-node": "^10.7.0",
     "typescript": "^4.6.4"
   },
   "scripts": {
     "start": "node dist/index.js",
     "build": "tsc",
-    "dev": "ts-node src/index.ts"
+    "dev": "nodemon src/index.ts"
```
Comment (Sparkenstein) [inline package.json:21]: we aren't using javascript we are using typescript, ts-node setup is already done use that

File: src/Routes/auth-login/auth-login.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
+require("dotenv").config();
+const { Router } = require("express");
```
Comment (Sparkenstein) [inline src/Routes/auth-login/auth-login.ts]: use esm don't use cjs

File: src/Routes/auth-login/auth-login.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+const {SESS_NAME = "sessionid",JWT_ACCESS_KEY="shivam"} = process.env
+
+const Users = [
+  {id:1,name:"shivam",email:"shivam@gmail.com",password:'password'},{id:2,name:"shiva",email:"shiva@gmail.com",password:'password'},{id:3,name:"pandey",email:"pandey@gmail.com",password:'password'},{id:4,name:"aditya",email:"aditya@gmail.com",password:'password'}
+]
+
+
+
+const authenicatelogin = (req: { session: any; },res: any,next: any) => {
```
Comment (Sparkenstein) [inline src/Routes/auth-login/auth-login.ts]: middlewares should be in their own folder

File: src/Routes/auth-login/auth-login.ts [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+  console.log(req.session.id)
+  if (!req.session.isauth) {
+    res.redirect("/login");
+  }
+  else { 
+    next()
+  }
+}
+
+const authenicatehome = (req: { session: any; },res: any,next: any) => {
```
Comment (Sparkenstein) [inline src/Routes/auth-login/auth-login.ts]: camerCase

File: src/Routes/auth-login/auth-login.ts [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    res.redirect("/home");
+  }
+  else { 
+    next()
+  }
+}
+
+loginauth.get("",(req: any,res: { send: (arg0: string) => any; }) => {
+    const {id} = req.session;
+    return res.send(`
```
Comment (Sparkenstein) [inline src/Routes/auth-login/auth-login.ts:20]: API wont' return any UI

File: src/Routes/auth-login/auth-login.ts [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+  return res.send(`
+    <h1>Welcome to Home Page</h1>
+  `)
+
+})
+
+loginauth.get("/login",authenicatehome,(req: any,res: any) => {
+   return res.send(`
+     <h1>Login</h1>
```
Comment (Sparkenstein) [inline src/Routes/auth-login/auth-login.ts:40]: API wont return any UI

File: src/Routes/auth-login/auth-login.ts [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+  req.session.destroy(error => {
+    if (error) {
+      return res.redirect("/home")
+    }
+    res.clearCookie(SESS_NAME);
+    res.redirect("/login");
+  })
+})
+
+export {loginauth}
```
Comment (Sparkenstein) [inline src/Routes/auth-login/auth-login.ts:133]: Everything is in file please separate out the features based on types

File: tsconfig.json [hunk 9]
```diff
@@ -74,7 +74,7 @@
     "forceConsistentCasingInFileNames": true /* Ensure that casing is correct in imports. */,
 
     /* Type Checking */
-    "strict": true /* Enable all strict type-checking options. */,
+    "strict": false /* Enable all strict type-checking options. */,
```
Comment (Sparkenstein) [inline tsconfig.json]: never change tsconfig

File: src/Routes/auth-login/auth-login.ts (hunk 7 above)
Comment (ShivCodeP) [reply src/Routes/auth-login/auth-login.ts:40]: ok I will remove this

File: src/Routes/auth-login/auth-login.ts (hunk 8 above)
Comment (ShivCodeP) [reply src/Routes/auth-login/auth-login.ts:133]: ok sure for middlewares can I create a directory with name middleware?

File: tsconfig.json (hunk 9 above)
Comment (ShivCodeP) [reply tsconfig.json]: fine

File: src/Routes/auth-login/auth-login.ts (hunk 6 above)
Comment (ShivCodeP) [reply src/Routes/auth-login/auth-login.ts:20]: ok I will remove this.

File: src/Routes/auth-login/auth-login.ts (hunk 4 above)
Comment (ShivCodeP) [reply src/Routes/auth-login/auth-login.ts]: for middlewares can I create a separate directory?

**Comment by Sparkenstein** [comment]
Closing it as inactive

**Pull request closed without merging**